taichi/transforms/optimize_bit_struct_stores.cpp

```diff
@@ -62,27 +62,31 @@ class MergeBitStructStores : public BasicStmtVisitor {
     auto &statements = block->statements;
     std::unordered_map<Stmt *, std::vector<BitStructStoreStmt *>>
         ptr_to_bit_struct_stores;
+    std::unordered_set<Stmt *> loaded_after_store;
     std::vector<Stmt *> statements_to_delete;
     for (int i = 0; i <= (int)statements.size(); i++) {
       // TODO: in some cases BitStructStoreStmts across container statements can
       // still be merged, similar to basic block v.s. CFG optimizations.
       if (i == statements.size() || statements[i]->is_container_statement()) {
-        for (const auto &item : ptr_to_bit_struct_stores) {
-          auto ptr = item.first;
-          auto stores = item.second;
-          if (stores.size() == 1)
+        for (const auto &[ptr, stores] : ptr_to_bit_struct_stores) {
+          if (stores.size() == 1) {
             continue;
-          std::map<int, Stmt *> values;
+          }
+
+          std::map<int, std::vector<Stmt *>> values;
           for (auto s : stores) {
             for (int j = 0; j < (int)s->ch_ids.size(); j++) {
-              values[s->ch_ids[j]] = s->values[j];
+              values[s->ch_ids[j]].push_back(s->values[j]);
             }
           }
+
           std::vector<int> ch_ids;
           std::vector<Stmt *> store_values;
-          for (auto &ch_id_and_value : values) {
-            ch_ids.push_back(ch_id_and_value.first);
-            store_values.push_back(ch_id_and_value.second);
+          for (auto const &[ch_id, store_value_vec] : values) {
```

Comment:
Looks like `values`'s only used here, do you think we can merge these two for-loops and get rid of `values`?